### PR TITLE
feat: support configuring idle timeout for generic commands [DET-4589]

### PR DESCRIPTION
**Pull request by shiyuann**
## Description

Support configuring idle timeout for generic commands. This supports TensorBoard and will support Notebook in a followup PR. 

## Test Plan

- Manually test it with TensorBoards

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/release-notes/2773-work-dir.txt
Comment (stoksc) [inline docs/release-notes/2773-work-dir.txt:5]: @shiyuann is this PR blocked on another (since title only describes the other feature) or am I to review both? either is fine just curious

Comment (shiyuann) [reply docs/release-notes/2773-work-dir.txt:5]: It's not blocked by anyone. It's to configure TensorBoard idle timeout, which we already support. But this will also help if we support Notebook idle timeout.

Comment (shiyuann) [reply docs/release-notes/2773-work-dir.txt:5]: But yes, it will be rebased if #2773 is merged. You can review the last two commits.

File: master/internal/task/idle_timeout.go
Comment (stoksc) [inline master/internal/task/idle_timeout.go:47]: wait, wouldn't asking yourself anything deadlock immediately? but you dont `.Get()` the resp it why it works. why not just a tell?

File: master/internal/task/idle_timeout.go
Comment (stoksc) [inline master/internal/task/idle_timeout.go:29]: can you put `actor.Addr("proxy")` into a pkg somewhere as a `var proxyAddr = ...` at the top. I think we should do this for all "well known" addresses (rm, trial logger, etc)

File: master/internal/task/idle_timeout.go
Comment (stoksc) [inline master/internal/task/idle_timeout.go:46]: alternative to passing description+kill message, what i've been doing is passing a `func(*actor.Context)` that encapsulates all the state to take a generic action in a subcomponent.

File: master/pkg/model/environment_config.go
Comment (stoksc) [inline master/pkg/model/environment_config.go:27]: appreciate the restructuring, maybe drop comments to describe the mini-sections within the struct?

File: master/pkg/tasks/task_command.go
Comment (stoksc) [inline master/pkg/tasks/task_command.go:27]: is this var accurately named?

**Review by stoksc — COMMENTED** [review]
mostly lgtm, nice refactor, definitely moving in the right direction

File: master/internal/task/idle_timeout.go
Comment (stoksc) [reply master/internal/task/idle_timeout.go:29]: or do it later/make a ticket for this pattern if you agree with it

File: master/internal/task/idle_timeout.go
Comment (shiyuann) [reply master/internal/task/idle_timeout.go:47]: good catch. Change it to tell

File: master/internal/task/idle_timeout.go
Comment (shiyuann) [reply master/internal/task/idle_timeout.go:29]: agreed. Will do that when we refactor interactive jobs.

File: master/pkg/tasks/task_command.go
Comment (shiyuann) [reply master/pkg/tasks/task_command.go:27]: Changed it to `WatchProxyIdleTimeout`

File: master/pkg/model/environment_config.go
Comment (shiyuann) [reply master/pkg/model/environment_config.go:27]: This is done in a different PR. Also, I think it's not very necessary to add comments for mini-sections since they are already self-explained.

File: master/internal/api_command.go
Comment (stoksc) [inline master/internal/api_command.go:136]: sorry, I'm a bit confused, the case where `config.Environment.WorkDir != nil && *workDirInDefaults == *config.Environment.WorkDir` is ok?

Comment (stoksc) [reply master/internal/api_command.go:136]: i guess my question is, why is this different than the check for experiments? sorry i'm out of the loop product side on this one.

**Review by stoksc — APPROVED** [review]
approved (obviously after conflicts/tests pass etc)

File: master/internal/api_command.go
Comment (stoksc) [reply master/internal/api_command.go:136]: i can't read

File: master/internal/command/command.go
Comment (stoksc) [inline master/internal/command/command.go:202]: fwiw, for this annoyance there is `ptrs.TimePtr()` that rb added. also, should this be called like "SetIdleness" ha.. Idk can't think of a great name for it.

**Review by stoksc — APPROVED** [review]

File: master/internal/command/command.go
Comment (stoksc) [inline master/internal/command/command.go:199]: my only real critique is that I may call this "IdleTaskRequest" or "IdleAllocationRequest" instead.

Comment (shiyuann) [reply master/internal/command/command.go:199]: This is only under `/api/v1/notebooks/` now and internal. I think it's fine to do so for now.

File: master/internal/command/command.go
Comment (shiyuann) [reply master/internal/command/command.go]: Let's make the names consistent when we refactor this. I think it's fine for now.

File: master/internal/command/command.go
Comment (stoksc) [inline master/internal/command/command.go:73]: I should've caught this in review @shiyuann but here we have unprotected concurrent access of cmd.lastActivity since it's captured by closure. We're very unlikely to hit it, but i'll fix it in the refactor I'm doing right now for task unif.

Comment (stoksc) [reply master/internal/command/command.go:73]: oh wait, it's not a different actor, nvm. i forgot it just has things that look like actor methods.

**Pull request merged**
